### PR TITLE
Update Recibo507 with Due Date

### PR DESCRIPTION
**Pull request by guilleJB**

### CODE REVIEW

File: bankbarcode/cuaderno57.py [hunk 1]
```diff
@@ -83,7 +110,8 @@ class Recibo507(Recibo):
         Autoservicio, V2001)
     """
 
-    def __init__(self, entity, suffix, ref, notice, amount):
+    def __init__(self, entity=None, suffix=None, ref=None, notice=None,
```
Comment (guilleJB) [inline bankbarcode/cuaderno57.py:113]: Update **init** with named arguments

File: bankbarcode/cuaderno57.py [hunk 2]
```diff
@@ -76,6 +77,32 @@ def _check_amount(self, amount):
...
+        Check due date (Fecha limite) and suffix
+
+        :param due_date: Due Date (Fecha limite)
+        :return: True if due date is a Datetime or string with format '%Y-%m-%d'
+                 and suffix bigger than 499, otherwise False
+        """
+        name = 'due_date'
+        if due_date is None:
+            return True
+        if isinstance(due_date, str):
```
Comment (ecarreras) [inline bankbarcode/cuaderno57.py]: Use `basestring` not `str` because `unicode` is not `str`.

Comment (guilleJB) [inline bankbarcode/cuaderno57.py]: Thanks

**Comment by ecarreras** [comment]
:+1:

File: bankbarcode/cuaderno57.py (hunk 1 above)
Comment (sim6) [inline bankbarcode/cuaderno57.py:113]: If I remember well this arguments are required, so if you named it we need to add a constrain.

**Pull request merged**